flask_session_cookie_manager.py

```diff
@@ -0,0 +1,108 @@
+""" Flask Session Cookie Decoder/Encoder """
+__author__ = 'Wilson Sumanang, Alexandre ZANNI'
+
+# standard imports
+import sys
+import zlib
+from itsdangerous import base64_decode
+import ast
+
+# for OOP
+from abc import ABC, abstractmethod
+
+# Lib for argument parsing
+import argparse
+
+# external Imports
+from flask.sessions import SecureCookieSessionInterface
+
+class MockApp(object):
+
+	def __init__(self, secret_key):
+		self.secret_key = secret_key
+
+
+class FSCM(ABC):
+	def encode(secret_key, session_cookie_structure):
+		""" Encode a Flask session cookie """
+		try:
+			app = MockApp(secret_key)
+
+			session_cookie_structure = dict(ast.literal_eval(session_cookie_structure))
+			si = SecureCookieSessionInterface()
+			s = si.get_signing_serializer(app)
+
+			return s.dumps(session_cookie_structure)
+		except Exception as e:
+			return "[Encoding error] {}".format(e)
+			raise e
+
+
+	def decode(session_cookie_value, secret_key=None):
+		""" Decode a Flask cookie  """
+		try:
+			if(secret_key==None):
+				compressed = False
+				payload = session_cookie_value
+
+				if payload.startswith("."):
```

Comment:
It's better to keep using single quotes rather than double quotes if no interpolation is needed. It changes nothing in Python but in other languages there is a difference, so let's use this as a convention.

Reply:
changed, thanks. Though I had in mind that the cookie prefix can possibly start with 'String' longer than 1 (i.e. a character), probably need to research further